src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -552,14 +561,29 @@ def exec_bulk(
             aggr_results=aggr_results,
         )
 
-    def validate_and_apply_inputs_mapping(self, inputs, inputs_mapping):
+    @staticmethod
+    def _apply_default_value_for_input(
+        inputs: Dict[str, FlowInputDefinition], line_inputs: Mapping, is_aggregation=False
+    ) -> Dict[str, Any]:
+        updated_inputs = dict(line_inputs or {})
+        for key, value in inputs.items():
+            if key not in updated_inputs and (value and value.default):
+                updated_inputs[key] = value.default if not is_aggregation else [value.default]
+        return updated_inputs
+
+    def validate_and_apply_inputs_mapping(self, inputs, inputs_mapping) -> List[Dict[str, Any]]:
```

Comment:
Will executor raise error when input not provided in `inputs_mapping` but has default value?

Reply:
Can we do the `_apply_default_value_for_input` here since control plane SDK are leveraging this function to validate inputs mapping

Reply:
executor will not raise error when input not provided in `inputs_mapping` but has default value.

Reply:
Will applying default in `validate_and_apply_inputs_mapping` is better than `exec_bulk`?
Applying default value seems like another thing from inputs mapping, don't have concrete reason to merge them. And we may need to change one better interface name.

Reply:
Offline synced, we don't have to apply default value in `validate_and_apply_inputs_mapping `